middleware/ordered_group.go

```diff
@@ -21,28 +21,51 @@ type ider interface {
 type orderedIDs struct {
 	order relativeOrder
 	items map[string]ider
-
-	unamedCounter int
+	slots map[string]struct{}
 }
 
 func newOrderedIDs() *orderedIDs {
 	return &orderedIDs{
 		items: map[string]ider{},
+		slots: map[string]struct{}{},
 	}
 }
 
+func (g *orderedIDs) IsSlot(id string) bool {
+	_, ok := g.slots[id]
+	return ok
+}
+
 // Add injects the item to the relative position of the item group. Returns an
 // error if the item already exists.
 func (g *orderedIDs) Add(m ider, pos RelativePosition) error {
-	if len(m.ID()) == 0 {
+	id := m.ID()
+	if len(id) == 0 {
 		return fmt.Errorf("empty ID, ID must not be empty")
 	}
 
-	if err := g.order.Add(m.ID(), pos); err != nil {
+	if !g.IsSlot(id) {
+		if err := g.order.Add(id, pos); err != nil {
+			return err
+		}
+	}
+
+	g.items[id] = m
+	return nil
```

Comment:
Should this check to see if a middleware already exists at the slot? It looks like this logic will allow a middleware to overwrite another middleware as long as it targets a slot. I'd expect this to return an error with the an `id` value already exists in `g.items`.